content/modules/r/pages/merge.adoc

```diff
@@ -0,0 +1,302 @@
+----
+books
+----
+
+[source, R]
+----
+   id                                     title author_id rating
+1   1     Harry Potter and the Sorcerers Stone          1   4.47
+2   2   Harry Potter and the Chamber of Secrets         1   4.43
+3   3  Harry Potter and the Prisoner of Azkaban         1   4.57
+4   4       Harry Potter and the Goblet of Fire         1   4.56
+5   5 Harry Potter and the Order of the Phoenix         1   4.50
+6   6    Harry Potter and the Half Blood Prince         1   4.57
+7   7      Harry Potter and the Deathly Hallows         1   4.62
+8   8                          The Way of Kings         2   4.64
+9   9                            The Book Thief         3   4.37
+10 10                      The Eye of the World         4   4.18
+----
+
+[source, R]
+----
+authors
+----
+
+[source, R]
+----
+   id                     name avg_rating
+1   1             J.K. Rowling       4.46
+2   2        Brandon Sanderson       4.39
+3   3             Markus Zusak       4.34
+4   4            Robert Jordan       4.18
+5   5          Agatha Christie       4.00
+6   6                Alex Kava       4.02
+7   7    Nassim Nicholas Taleb       3.99
+8   8              Neil Gaiman       4.13
+9   9            Stieg Larsson       4.16
+10 10 Antoine de Saint-Exupéry       4.30
+----
+
+==== How do I merge the author information from `authors` based on `author_id` in `books` and `id` in `authors`, keeping only information from authors and books where there is a match?
+
+.Click to see solution
+[%collapsible]
+====
+[source, R]
+----
+# In SQL this is referred to as an INNER JOIN.
+merge(books, authors, by.x="author_id", by.y="id", all=F)
+----
+
+[source, R]
+----
+   author_id id                                     title rating
+1          1  1     Harry Potter and the Sorcerers Stone    4.47
+2          1  2   Harry Potter and the Chamber of Secrets   4.43
+3          1  3  Harry Potter and the Prisoner of Azkaban   4.57
+4          1  4       Harry Potter and the Goblet of Fire   4.56
+5          1  5 Harry Potter and the Order of the Phoenix   4.50
+6          1  6    Harry Potter and the Half Blood Prince   4.57
+7          1  7      Harry Potter and the Deathly Hallows   4.62
+8          2  8                          The Way of Kings   4.64
+9          3  9                            The Book Thief   4.37
+10         4 10                      The Eye of the World   4.18
+                name avg_rating
+1       J.K. Rowling       4.46
+2       J.K. Rowling       4.46
+3       J.K. Rowling       4.46
+4       J.K. Rowling       4.46
+5       J.K. Rowling       4.46
+6       J.K. Rowling       4.46
+7       J.K. Rowling       4.46
+8  Brandon Sanderson       4.39
+9       Markus Zusak       4.34
+10     Robert Jordan       4.18
+----
+====
+
+==== How do I merge the author information from `authors` based on `author_id` in `books` and `id` in `authors`, keeping all information from `authors` regardless of whether or not there is match?
```

Comment:
How do I merge both data.frames based on the `author_id` column in the `books` data and the `id` column in the `authors` data, keeping **all** rows regardless of whether or not there is a match?